hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SpecificPartitionsImapReaderPmsTest.java

```diff
@@ -22,110 +22,233 @@
-        map.put(0, 0);
+        sourceMap.put(0, 0);
 
         DAG dag = new DAG();
-        ProcessorMetaSupplier readPms = mapReader(mapName, null, List.of(
-                List.of(ConstantExpression.create(0, QueryDataType.INT))));
+        ProcessorMetaSupplier readPms = mapReader(mapName, null, null);
         Vertex source = dag.newVertex("source", readPms);
         Vertex sink = dag.newVertex("sink", writeMapP(sinkName));
 
         dag.edge(between(source, sink));
         instance().getJet().newLightJob(dag, jobConfig).join();
 
         assertEquals(0, instance().getMap(sinkName).get(0));
+
+        Vertex sourceV = dag.getVertex("source");
+        assertNotNull(sourceV);
+
+        ProcessorMetaSupplier metaSupplier = sourceV.getMetaSupplier();
+        assertNotNull(metaSupplier);
+        assertInstanceOf(SpecificPartitionsImapReaderPms.class, metaSupplier);
+
+        SpecificPartitionsImapReaderPms pms = (SpecificPartitionsImapReaderPms) metaSupplier;
+        // Ensure we scan all partitions.
+        assertNull(pms.partitionsToScan);
     }
 
-    @SuppressWarnings("rawtypes")
     @Test
-    public void test_prunable() {
-        String mapName = "m";
+    public void test_prunableSinglePartition() {
+        // Given
+        int partitionsToUse = 1;
         IMap<Integer, Integer> map = instance().getMap(mapName);
-        map.put(pKey, pKey);
+        DAG dag = setupPrunableDag(sourceMap, partitionsToUse);
 
+        // When
+        instance().getJet().newLightJob(dag, jobConfig).join();
+
+        // Then
+        assertPrunability(partitionsToUse);
+    }
+
+    @Test
+    public void test_prunableMultiplePartitions() {
         // Given
-        ProcessorMetaSupplier pms = mapReader(mapName, null, List.of(
-                List.of(ConstantExpression.create(pKey, QueryDataType.INT))
-        ));
+        int partitionsToUse = 3;
+        DAG dag = setupPrunableDag(sourceMap, partitionsToUse);
 
         // When
-        // Note: processor verification support is bounded to one member only,
-        //  so it is applicable to make this test in 'processor verification' way.
-        TestSupport.verifyProcessor(adaptSupplier(pms))
-                .hazelcastInstance(instance())
-                .jobConfig(jobConfig)
-                .disableSnapshots()
-                .disableProgressAssertion()
-                .expectExactOutput(out(entry(pKey, pKey)));
+        instance().getJet().newLightJob(dag, jobConfig).join();
 
         // Then
-        Assert.assertEquals(1, ((SpecificPartitionsImapReaderPms) pms).partitionsToScan.length);
-        Assert.assertEquals(keyOwnerPartitionId, ((SpecificPartitionsImapReaderPms) pms).partitionsToScan[0]);
+        assertPrunability(partitionsToUse);
+    }
+
+    private DAG setupPrunableDag(IMap<Integer, Integer> map, int partitionCountToUse) {
+        assertGreaterOrEquals("Should scan at least one partition", partitionCountToUse, 1);
```

Comment:
is empty partitions list invalid for SpecificPartitionsImapReaderPms? There is no such validation and it should work just fine in that case.
We should test edge cases.

Reply:
That method is used when we are setting up a case where prunability **must** work. You may notice in `test_basic`, that it tests scanning for basic scenario without declared partitioned strategy.

Reply:
unit tests should check also boundary conditions. Added such check.